crates/ruff/src/checkers/tokens.rs

```diff
@@ -184,9 +192,39 @@ pub(crate) fn check_tokens(
     }
 
     // TD001, TD002, TD003, TD004, TD005, TD006, TD007
+    // T001, T002, T003, T004
     if enforce_todos {
+        // The TextRange of the comment, its position in comment_ranges, and the directive's
+        // TextRange
+        let mut other_directive_ranges: Vec<(TextRange, usize, TextRange)> = vec![];
```

Comment:
Nit: I think it would be good to define a new struct for `(TextRange, usize, TextRange)`  to have a name for the type and its field `CommentDirective` (I find tuples with more than two elements hard to read, particularly if they have multiple fields of the same type)